src/service/scenarios/delete_scenarios_service.py

```diff
@@ -0,0 +1,237 @@
+        if not from_details:
+            for scenario in scenarios:
+                print(scenario)
+                scenario_is_deleted = self.delete_scenario(
+                    scenario["name"],
+                    scenario["year"],
+                    company_id,
+                    scenario["metric"],
+                    scenario["value"],
+                )
+                if scenario_is_deleted:
+                    scenarios_deleted += 1
+        else:
+            for scenario in scenarios:
+                scenario_is_deleted = self.delete_scenario_from_details(
+                    scenario["scenario_id"], scenario["metric_id"]
+                )
+                if scenario_is_deleted:
+                    scenarios_deleted += 1
+
+        return scenarios_deleted
+
+    def delete_currency(self, metric_id: str) -> bool:
+        try:
+            query = """
+            DELETE FROM {table_name} WHERE metric_id = '{metric_id}'
+            """.format(
+                table_name="currency_metric", metric_id=metric_id
+            )
+
+            self.session.execute(query)
+            self.session.commit()
+            return True
+        except Exception as error:
+            self.logger.info(error)
+            return False
+
+    def delete_scenario_metric(self, scenario_id: str, metric_id: str) -> bool:
+        currency_is_deleted = self.delete_currency(metric_id)
+
+        if currency_is_deleted:
+            try:
+                query = """
+                    DELETE FROM {table_name}
+                    WHERE scenario_id = '{scenario_id}'
+                    AND metric_id = '{metric_id}'
+                """.format(
+                    table_name="scenario_metric",
+                    scenario_id=scenario_id,
+                    metric_id=metric_id,
+                )
+
+                self.session.execute(query)
+                self.session.commit()
+                return True
+            except Exception as error:
+                self.logger.info(error)
+                return False
+        else:
+            return False
+
+    def delete_metric(self, scenario_id: str, metric_id: str) -> bool:
+        scenario_metric_is_deleted = self.delete_scenario_metric(scenario_id, metric_id)
+        if scenario_metric_is_deleted:
+            try:
+                query = """
+                DELETE FROM {table_name} WHERE id = '{metric_id}'
+                """.format(
+                    table_name="metric", metric_id=metric_id
+                )
+
+                self.session.execute(query)
+                self.session.commit()
+                return True
+            except Exception as error:
+                self.logger.info(error)
+                return False
+        else:
+            return False
+
+    def delete_scenario_from_details(self, scenario_id: str, metric_id) -> bool:
+        metric_is_deleted = self.delete_metric(scenario_id, metric_id)
+        if metric_is_deleted:
```

Comment:
Don't too much if, try to validate the condition with the failed result, example if not metric_is_deleted: return False